File: app/src/main/java/com/example/newstart/MainActivity.java
package com.example.newstart;

import android.os.Bundle;

import androidx.annotation.Nullable;

import com.example.newstart.databinding.ActivityMainBinding;

public class MainActivity extends BaseActivity {
    private ActivityMainBinding mBinding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(mBinding.getRoot());
        addListener();
    }

    private void addListener() {
        mBinding.btnConstraintLayout.setOnClickListener(v -> ConstrainLayoutActivity.actionStart(mContext));
        mBinding.btnCustomDrawable.setOnClickListener(v -> CustomDrawableActivity.actionStart(mContext));
        mBinding.btnCard3d.setOnClickListener(v -> Rotate3dActivity.actionStart(mContext));
        mBinding.btnXmClock.setOnClickListener(v -> XmViewActivity.actionStart(mContext));
        mBinding.btnCamera.setOnClickListener(v -> ThreeDirectionActivity.actionStart(mContext));
        mBinding.btnEventConflict.setOnClickListener(v -> EventConflictActivity.actionStart(mContext));
        mBinding.btnMultiTouch.setOnClickListener(v -> MultiTouchActivity.actionStart(mContext));
        mBinding.btnScroller.setOnClickListener(v -> ScrollerActivity.actionStart(mContext));
        mBinding.btnViewDrag.setOnClickListener(v -> ViewDragActivity.actionStart(mContext));
        mBinding.btnRecyclerView.setOnClickListener(v -> RecyclerViewActivity.actionStart(mContext));
        mBinding.btnAnim.setOnClickListener(v -> AnimActivity.actionStart(mContext));
        mBinding.btnHwClock.setOnClickListener(v -> HwClockActivity.actionStart(mContext));
        mBinding.btnPie.setOnClickListener(v -> PieActivity.actionStart(mContext));
        mBinding.btnVolume.setOnClickListener(v -> VolumeActivity.actionStart(mContext));
    }
}

File: kotlin/src/main/java/com/example/kotlin/MainActivity.kt
package com.example.kotlin

import android.os.Bundle
import androidx.activity.ComponentActivity
import com.example.kotlin.databinding.ActivityMainBinding
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.MainScope
import kotlinx.coroutines.launch

class MainActivity : ComponentActivity() {

    private val mBinding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }

    private val scope = MainScope()

    companion object {
        private val TAG = MainActivity::class.java.simpleName
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(mBinding.root)

        scope.launch(Dispatchers.Main){
            
        }
    }
    
    private suspend fun obtainCacheData(){

    }
}
